chore(ci): scripts/ is outside the lint scope — 7 deprecated-API errors sitting unlinted in the governance generators

The repo's lint script is `eslint src/` (`packages/nexus-agents/package.json`). **`scripts/` is never linted.**

That directory is not incidental — it holds `inject-governance.ts`, the drift gates (`check-harness-alignment`, `check-mcp-description-drift`, `check-authority-tier-drift`), `generate-tool-reference`, `generate-skills-index`, the release helper `changeset-version-with-retry`, and `verify-refresh.sh`'s TypeScript siblings. **The code that enforces the repo's rules is itself unenforced.**

## What is sitting there right now

Running eslint over `scripts/` surfaces 7 errors, all deprecated Zod APIs in `scripts/build-model-registry-types.ts`:

```
 43:8   error  `passthrough` is deprecated. Use `z.looseObject()` or `.loose()` instead
 52:8   error  (same)
 60:8   error  (same)
 63:4   error  (same)
 81:4   error  (same)
116:4   error  (same)
143:27  error  `url` is deprecated. Use `z.url()` instead
```

Plus 2 `no-console` warnings in `scripts/sync-plugin-version.ts`.

Confirmed **pre-existing** — they reproduce with recent dependency bumps stashed, so they are not fallout from #4482.

## Why it matters beyond the 7

This is the same shape as two other defects found today:

- **#4473** — no CI job exercises `release.yml`, so a release-breaking action bump merged green
- The `brace-expansion` override that silently stopped matching

In each case a check *appeared* to cover something it did not. Per the #4462 north star: *a check that cannot fail by construction is not a check.* An unlinted directory cannot report a lint failure.

## Scope

- [ ] Extend the lint scope to cover `scripts/` (and confirm whether `website/` and `packages/nexus-memory/src` are in scope — `nexus-memory` appears to be, `website` uses `astro check` separately)
- [ ] Fix the 7 Zod deprecations: `.passthrough()` → `.looseObject()`, `z.string().url()` → `z.url()`
- [ ] Decide on the 2 `no-console` warnings — a build script legitimately prints, so an override for `scripts/**` may be the right answer rather than silencing individually
- [ ] Land the scope extension and the fixes as **separate** commits, so the newly-covered surface is auditable

## Sequencing note

Extend the scope *first* and let it fail, then fix — so the gate is demonstrated to work before the errors disappear. Fixing first and extending after would leave no evidence the scope change actually catches anything.

--- a/.changeset/lint-scripts-scope.md
+++ b/.changeset/lint-scripts-scope.md
@@ -1,0 +1,13 @@
+---
+'nexus-agents': patch
+---
+
+Bring the repo-root `scripts/` tree into the lint scope (#4483).
+
+`pnpm lint` is `turbo lint`, which only reaches each workspace's own `eslint src/`. The root `scripts/` directory — governance generators, drift gates, the tool-reference generator, the release helper — was outside every lint scope. The code that enforces the repo's rules was itself unenforced, and had accumulated 7 deprecated-API errors nothing reported.
+
+Adds `lint:scripts` and a CI step that runs it. The scope was extended **first** and confirmed to fail with all 7 errors before anything was fixed, so the gate is demonstrated to catch something rather than landing already-green.
+
+Fixes the 7: `.passthrough()` → `.loose()` and `z.string().url()` → `z.url()` in `build-model-registry-types.ts`. Verified behaviour-preserving by running the generator — it exits clean and produces a byte-identical registry, so this is a rename, not a semantic change.
+
+`sync-plugin-version.ts` gets the file-level `no-console` disable its 38 sibling scripts already carry; a build script's stdout is its interface.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -75,6 +75,13 @@ jobs:
       - name: Run lint
         run: pnpm lint
 
+      # #4483: `pnpm lint` is `turbo lint`, which only reaches each workspace's
+      # own `eslint src/`. The repo-root `scripts/` tree — governance
+      # generators, drift gates, the release helper — was outside every lint
+      # scope and had accumulated 7 deprecated-API errors nothing reported.
+      - name: Run lint (repo scripts)
+        run: pnpm lint:scripts
+
   typecheck:
     name: Type Check
     runs-on: ubuntu-latest

--- a/package.json
+++ b/package.json
@@ -33,6 +33,7 @@
     "test:scripts": "vitest run",
     "lint": "turbo lint",
     "lint:fix": "turbo lint:fix",
+    "lint:scripts": "eslint scripts/",
     "lint:arch": "npx tsx scripts/arch-lint.ts",
     "lint:md": "markdownlint 'docs/**/*.md' '*.md' --ignore 'docs/api/**' --ignore 'node_modules' --config .markdownlint.json",
     "lint:md:fix": "markdownlint 'docs/**/*.md' '*.md' --ignore 'docs/api/**' --ignore 'node_modules' --config .markdownlint.json --fix",

--- a/scripts/build-model-registry-types.ts
+++ b/scripts/build-model-registry-types.ts
@@ -7,7 +7,7 @@
  * generated registry pulls from upstream sources that cover hundreds of
  * providers and thousands of model ids we don't want to enumerate.
  *
- * `.passthrough()` is used on upstream shapes so a new field showing up in
+ * `.loose()` is used on upstream shapes so a new field showing up in
  * models.dev / LiteLLM doesn't break the build — it flows through and is
  * trimmed during output mapping.
  *
@@ -40,7 +40,7 @@ export const ModelsDevEntrySchema = z
         input: z.array(z.string()).optional(),
         output: z.array(z.string()).optional(),
       })
-      .passthrough()
+      .loose()
       .optional(),
     cost: z
       .object({
@@ -49,7 +49,7 @@ export const ModelsDevEntrySchema = z
         cache_read: z.number().nonnegative().optional(),
         cache_write: z.number().nonnegative().optional(),
       })
-      .passthrough()
+      .loose()
       .optional(),
     limit: z
       .object({
@@ -57,10 +57,10 @@ export const ModelsDevEntrySchema = z
         input: z.number().int().nonnegative().optional(),
         output: z.number().int().nonnegative().optional(),
       })
-      .passthrough()
+      .loose()
       .optional(),
   })
-  .passthrough();
+  .loose();
 
 export type ModelsDevEntry = z.infer<typeof ModelsDevEntrySchema>;
 
@@ -78,7 +78,7 @@ export const ModelsDevProviderSchema = z
     npm: z.string().optional(),
     models: z.record(z.string(), z.unknown()),
   })
-  .passthrough();
+  .loose();
 
 export type ModelsDevProvider = z.infer<typeof ModelsDevProviderSchema>;
 
@@ -113,7 +113,7 @@ export const LiteLlmEntrySchema = z
     supports_pdf_input: z.boolean().optional(),
     deprecation_date: z.string().optional(),
   })
-  .passthrough();
+  .loose();
 
 export type LiteLlmEntry = z.infer<typeof LiteLlmEntrySchema>;
 
@@ -140,7 +140,7 @@ export const MAX_UPSTREAM_PAYLOAD_BYTES = 5 * 1024 * 1024;
 export const GeneratedProvenanceSchema = z.object({
   source: z.enum(['models.dev', 'litellm']),
   fetchedAt: z.string(),
-  upstreamUrl: z.string().url(),
+  upstreamUrl: z.url(),
 });
 
 export type GeneratedProvenance = z.infer<typeof GeneratedProvenanceSchema>;

--- a/scripts/sync-plugin-version.ts
+++ b/scripts/sync-plugin-version.ts
@@ -7,6 +7,8 @@
  * @module scripts/sync-plugin-version
  */
 
+/* eslint-disable no-console -- build script; its stdout IS the interface (#4483) */
+
 import { readFileSync, writeFileSync } from 'node:fs';
 import { resolve } from 'node:path';
 

